upgraderesponder/service.go

```diff
@@ -18,25 +18,34 @@ import (
 )
 
 const (
-	VersionTagLatest       = "latest"
-	LonghornMinimalVersion = "v0.0.1"
-
-	// ns is good for counting nodes
-	InfluxDBPrecisionNanosecond = "ns"
-	InfluxDBDatabase            = "longhorn_upgrade_responder"
-
-	InfluxDBMeasurementName = "longhorn_upgrade_query"
+	VersionTagLatest  = "latest"
+	AppMinimalVersion = "v0.0.1"
+
+	InfluxDBMeasurement                    = "upgrade_request"
+	InfluxDBMeasurementDownSampling        = "upgrade_request_down_sampling"
+	InfluxDBMeasurementByAppVersion        = "by_app_version_down_sampling"
+	InfluxDBMeasurementByKubernetesVersion = "by_kubernetes_version_down_sampling"
+	InfluxDBMeasurementByCountryCode       = "by_country_code_down_sampling"
+
+	InfluxDBContinuousQueryDownSampling        = "cq_upgrade_request_down_sampling"
+	InfluxDBContinuousQueryByAppVersion        = "cq_by_app_version_down_sampling"
+	InfluxDBContinuousQueryByKubernetesVersion = "cq_by_kubernetes_version_down_sampling"
+	InfluxDBContinuousQueryByCountryCode       = "cq_by_country_code_down_sampling"
 )
 
 var (
-	InfluxDBTagLonghornVersion        = "longhorn_version"
+	InfluxDBPrecisionNanosecond   = "ns" // ns is good for counting nodes
+	InfluxDBDatabase              = "upgrade_responder"
+	InfluxDBContinuousQueryPeriod = "1h"
+
+	InfluxDBTagAppVersion             = "app_version"
 	InfluxDBTagKubernetesVersion      = "kubernetes_version"
```

Comment:
Does `kubernetes_version` make sense to all the applications?

Reply:
Yeah, `kubernetes_version` doesn't make sense for all the applications. If the application doesn't send `kubernetes_version`. The Kubernetes graph will be the same as the total node count graph in Grafana. Users can just directly delete the Kubernetes graph